bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeConverter.java

```diff
@@ -262,6 +331,57 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
         return convertibleTypes;
     }
 
+    private static boolean isConvertibleToRecordType_bstring(Object sourceValue, BRecordType targetType,
+                                                             List<TypeValuePair> unresolvedValues) {
+        if (!(sourceValue instanceof MapValueImpl)) {
+            return false;
+        }
+
+        TypeValuePair typeValuePair = new TypeValuePair(sourceValue, targetType);
+        if (unresolvedValues.contains(typeValuePair)) {
+            return true;
+        }
+        unresolvedValues.add(typeValuePair);
+
+        Map<String, BType> targetFieldTypes = new HashMap<>();
+        BType restFieldType = targetType.restFieldType;
+
+        for (BField field : targetType.getFields().values()) {
+            targetFieldTypes.put(field.getFieldName(), field.type);
+        }
+
+        MapValueImpl sourceMapValueImpl = (MapValueImpl) sourceValue;
+        for (Map.Entry targetTypeEntry : targetFieldTypes.entrySet()) {
+            String fieldName = targetTypeEntry.getKey().toString();
+
+            if (!sourceMapValueImpl.containsKey(StringUtils.fromString(fieldName))) {
```

Comment:
inverse if condition here

Reply:
I inverted the condition, but this is also a copy of an existing function for bstring changes

Reply:
ack